azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/AuthHelper.java

```diff
@@ -61,7 +59,7 @@ public static boolean validateEnvironment(String environment) {
         if (StringUtils.isBlank(environment)) {
             return true;
         }
-        switch (environment.toLowerCase(Locale.ENGLISH)) {
+        switch (StringUtils.remove(environment.toLowerCase(Locale.ENGLISH), "_")) {
```

Comment:
may use `StringUtils.equalsAnyIgnoreCase`, btw, should we define the valid values here or values like `azurechin_a` will also returns true here

Reply:
we will need to support/accept all kinds of ways by user(eg: trim spaces, use '-', '_' , 'AzureCloud' or 'Azure'), I think I may change it by remove one time of '_', and '_' and 'Cloud' at end